app/scripts/directives/mobileServiceInstanceRow.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('mobileServiceInstanceRow', {
+    controller: [
+      '$filter',
+      '$routeParams',
+      '$scope',
+      'APIService',
+      'AuthorizationService',
+      'BindingService',
+      'Constants',
+      'DataService',
+      'ListRowUtils',
+      'ServiceInstancesService',
+      MobileServiceInstanceRow
+    ],
+    controllerAs: 'row',
+    bindings: {
+        apiObject: '<',
+        serviceClass: '<',
+        binding: '<',
+        state: '<'
+    },
+    templateUrl: 'views/_mobile-service-instance-row.html'
+  });
+
+  function MobileServiceInstanceRow($filter,
+                              $routeParams,
+                              $scope,
+                              APIService,
+                              AuthorizationService,
+                              BindingService,
+                              Constants,
+                              DataService,
+                              ListRowUtils,
+                              ServiceInstancesService) {
+    var row = this;
+    var serviceInstanceDisplayName = $filter('serviceInstanceDisplayName');
+    row.serviceBindingsVersion = APIService.getPreferredVersion('servicebindings');
+    row.state = row.state || {};
+    var annotationSpace = "org.aerogear.binding." + this.apiObject.metadata.name;
+    var watches = [];
+    _.extend(row, ListRowUtils.ui);
+
+    $scope.$on('$destroy', function(){
+        DataService.unwatchAll(watches);
+    });
+
+    watches.push(DataService.watchObject(Constants.MOBILE_CLIENT_VERSION, $routeParams.mobileclient, {namespace: $routeParams.project}, function(mobileClient, action) {
+        row.mobileClient = mobileClient;
+        row.annotations = [];
+        row.parameterData = {CLIENT_ID: row.mobileClient.metadata.name};
```

Comment:
How we are prepopulating the params is here:

https://github.com/aerogear/origin-web-console/commit/b4048e47814f8c7249dd15bdd91b42185562411e#diff-0572175f954d118f1869e17803496186R72

https://github.com/aerogear/origin-web-console/commit/b4048e47814f8c7249dd15bdd91b42185562411e#diff-0572175f954d118f1869e17803496186R129